llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -11542,22 +11542,47 @@ SDValue TargetLowering::expandFP_TO_INT_SAT(SDNode *Node,
   // If the integer bounds are exactly representable as floats and min/max are
   // legal, emit a min+max+fptoi sequence. Otherwise we have to use a sequence
   // of comparisons and selects.
-  bool MinMaxLegal = isOperationLegal(ISD::FMINNUM, SrcVT) &&
+  bool MinMax2019NumLegal = isOperationLegal(ISD::FMINIMUMNUM, SrcVT) &&
+                     isOperationLegal(ISD::FMAXIMUMNUM, SrcVT);
```

Comment:
probably should be isLegalOrCustom

Reply:
No. Normally the custom of FMAXIMUMNUM is quite expensive.
Fallback to fcmp is much cheaper.